src/pages/studyView/UserSelections.tsx

```diff
@@ -64,26 +66,26 @@ export default class UserSelections extends React.Component<IUserSelectionsProps
         let chartMeta = this.props.attributesMetaSet[UniqueKey.CNA_GENES_TABLE];
         if (chartMeta && !_.isEmpty(this.props.filter.cnaGenes)) {
             return (
-                <span className={styles.filter}>
-                    <span className={styles.name}>{chartMeta.displayName}</span>
-                    <i className="fa fa-times" style={{ cursor: "pointer" }} onClick={this.props.clearCNAGeneFilter}></i>
-                </span>
-            )
+                <ChartFilter
+                    chartMeta={chartMeta}
+                    clearFilter={this.props.clearCNAGeneFilter}
+                />)
         }
         return null;
     }
 
-    @computed private get mutationCountVScnaFilter() {
-        let chartMeta = this.props.attributesMetaSet[UniqueKey.MUTATION_COUNT_CNA_FRACTION];
-        if (chartMeta && !_.isEmpty(this.props.filter.sampleIdentifiers)) {
-            return (
-                <span className={styles.filter}>
-                    <span className={styles.name}>{chartMeta.displayName}</span>
-                    <i className="fa fa-times" style={{ cursor: "pointer" }} onClick={this.props.clearCustomCasesFilter}></i>
-                </span>
-            )
-        }
-        return null;
+    @computed private get sampleIdentifiersFilters() {
+        return _.map(this.props.filter.sampleIdentifiersSet, (sampleIdentifiers, chartUniqKey) => {
```

Comment:
lets use reduce instead of returning null values.  or filter then man if there aren't going to be that many items in this list and we don't care too much about performance.